### DISCUSSION TITLE
Bonsai crashing with multiple AudioCapture nodes

### DISCUSSION THREAD

**Discussion by aliplump** [bonsai-rx/bonsai#1698] [Q&A]
I have a program that simply has two AudioCapture nodes connected to channels from a Focusrite Scarlett 18i20. No matter what buffer size I choose, the app always crashes with a c0000374 exception code. Is there not capability to capture audio from multiple channels at once on Bonsai? Or am I doing something else wrong? I am fairly new to Bonsai, so open to any suggestions.
[audioCapture-test.zip](https://github.com/bonsai-rx/bonsai/files/14453920/audioCapture-test.zip)

**Comment by bruno-f-cruz** [comment]
I just made a quick test on my PC and I seem to be able to acquire from two separate mics at the same time. I suspect this might be a problem with your hardware/drivers configuration unfortunately :(

**Comment by BenjaminEscribano** [comment]
Hey,

I have the same issue (see: https://github.com/bonsai-rx/bonsai/issues/1693). We are using two Pettersson USB microphones and when stopping the acquisition, Bonsai will crash. It will save the .wav files but with some interference/noise in the audio file. I have tested the same with internal microphones and experienced the same issue. Everything was done with Bonsai 2.8.

**Comment by aliplump** [comment]
After digging through both the code and potential hardware problems, I think my issue stems from the disposal at the end of the using statement in Bonsai's AudioCapture code. The hardware I'm using (Focusrite Scarlett 18i20) is one device that splits into multiple channels. I'm thinking the issue is the Focusrite provides one reference for both channels even though they are separate in function, so AudioCapture node 1 closes the device after AudioCapture node 2 has already created a pointer to the same device. So when AudioCapture node 2 tries to close the device, it runs into the error that the device is closed already and shuts down the application.

I'm not certain this is exactly what is happening, or if this is applicable to the issue you're having @BenjaminEscribano but I think the best solution for me would be to switch devices. 

Let me know if anyone has other thoughts on this or if I got something wrong.
